v0/destinations/blueshift/transform.js

```diff
@@ -116,58 +113,29 @@ const identifyResponseBuilder = async (message, category, { Config }) => {
 };
 
 const groupResponseBuilder = async (message, category, { Config }) => {
-  if (!Config.usersApiKey) {
+  if (!Config.eventApiKey) {
     throw new CustomError(
-      "[BLUESHIFT] User API Key required for Authentication.",
+      "[BLUESHIFT] event API Key required for Authentication.",
       400
     );
   }
 
-  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
 
   if (!payload) {
     // fail-safety for developer error
     throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
   }
-  if (!payload.list_id) {
-    if (payload.name && payload.description) {
-      const baseURL = Config.datacenterEU ? BASE_URL_EU : BASE_URL;
-      const endpoint = `${baseURL}/api/v1/custom_user_lists/create`;
-
-      const basicAuth = Buffer.from(Config.usersApiKey).toString("base64");
-      const requestOptions = {
-        headers: {
-          "Content-Type": "application/json",
-          Authorization: `Basic ${basicAuth}`
-        }
-      };
-      const blueshiftlist = await httpPOST(endpoint, payload, requestOptions);
-      if (blueshiftlist.success === true && blueshiftlist.response.data.id) {
-        payload.list_id = blueshiftlist.response.data.id;
-      } else {
-        throw new CustomError(
-          `[Blueshift]:: ${blueshiftlist.response.message}`,
-          400
-        );
-      }
-    } else {
-      throw new CustomError(
-        "[Blueshift]:: name and description are required to create an empty list, or List Id is required to add user.",
-        400
-      );
-    }
-  }
+  payload.event = "identify";
+  payload = extractCustomFields(message, payload, ["traits"], ["cookie"]);
```

Comment:
Suggestion: As `cookie` is being mapped from traits only and it is **not** a mandatory field, we can simply move all the properties from traits here and do not need to add it in the mapping json.

Reply:
In mapping JSON the mapping is sourced from specific fields like `context.device.id` which will be difficult to implement using this function